### DISCUSSION TITLE
Undo/Redo Deletes Parts of Drawings?

### DISCUSSION THREAD

**Discussion by HyperAnimated** [keiwando/toonsquid#523] [Q&A] [answered]
Hi everyone!  Just had a bizarre experience and wanted to ask if this is a feature I screwed up, or if I just encountered a bug, please:

- Created a new project, traditional style.
- Drew one frame of an oval with a sketch pencil, no fill.  Next frame, drew a new oval up and to the right.  
- Turned on Onion Skinning.  Tried to use Off Peg for the first time, realized I didn't understand how it worked yet, turned it back off.
- Drew a third frame/oval, then a fourth.
- Decided I needed to move things down/left for better spacing.
- Selected all four frames on the timeline.
- Used arrow to try to move them all.  Only first frame moved.
- Tapped with two fingers on screen to undo.  Instead of moving it back, each time I tried to undo/redo, it began deleting seemingly random parts of all four frames I'd drawn, jumped part of one down to the lower right corner of the canvas, and finally left me with only a diagonal half of the original oval (which I'd drawn in one motion, so undo wouldn't have removed just half), and no more option to undo/redo.

Does anyone have any clue what happened?

**Comment by Cat-Butler** [comment]
I have no idea what is happening here, I personally haven't seen and can't reproduce this behaviour. The only time I I have seen occasional odd behaviour is with the Fill tool, interactive threshold, and multiple Undo/Redo.

However, it might be worth doing the usual "weird iPad behaviour solutions", on the chance that it is iPad/OS related, rather than ToonSquid related:
1. [Hard/forced restart](https://support.apple.com/en-au/guide/ipad/ipad9955c007/ipados).
2. Check free storage: iPad Settings app > General > iPad Storage. Once the amount of remaining storage gets down to under ... umm, can't remember exactly, maybe 15%, maybe less ... then iPad behaviour can start getting flaky (used for on-the-fly swapping of in-memory apps/data to storage, i.e. virtual memory, I think). If there isn't a lot of storage left, it might be time to consider deleting some apps or files.

**Comment by keiwando** [comment]
Definitely sounds like a bug. I'll be looking into it. Thanks for the report!

**Reply by keiwando** [reply, marked as answer]
This should be fixed now with the 1.10.7 update.

**Reply by HyperAnimated** [reply]
Thanks very much, I appreciate it! :)
